src/components/CreateTargetForm/index.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from 'react';
+import { func } from 'prop-types';
+import { useStatus, LOADING } from '@rootstrap/redux-tools';
+import useValidation from 'hooks/useValidation';
+import useForm from 'hooks/useForm';
+import useTextInputProps from 'hooks/useTextInputProps';
+import usePickerProps from 'hooks/usePickerProps';
+
+import Input from 'components/common/Input';
+import ErrorView from 'components/common/ErrorView';
+import TopicListModal from 'components/TopicListModal';
+import TopicPicker from 'components/common/TopicPicker';
+import Button from 'components/common/Button';
+import createTargetValidations from 'validations/createTargetValidations';
+import { createTarget } from 'actions/targetActions';
+import strings from 'localization';
+import { BLACK } from 'constants/colors';
+import styles from './styles';
+
+const FIELDS = {
+  areaLenght: 'areaLenght',
+  targetTitle: 'targetTitle',
+  topic: 'topic',
+};
+
+const CreateTargetForm = ({ onSubmit, userLocation }) => {
+  const [isTopicModalVisible, setIsTopicModalVisible] = useState(false);
+  const { error, status } = useStatus(createTarget);
+  const validator = useValidation(createTargetValidations);
+  const { latitude, longitude } = userLocation;
+
+  const {
+    values,
+    errors,
+    handleValueChange,
+    handleSubmit,
+    handleFocus,
+    handleBlur,
+    activeFields,
+    touched,
+    formHasErrors,
+  } = useForm(
+    {
+      onSubmit,
+      initialValues: { areaLenght: '200', latitude, longitude },
+      validator,
+      validateOnBlur: true,
+      validateOnChange: true,
+    },
+    [onSubmit],
+  );
+
+  const inputProps = useTextInputProps(
+    handleValueChange,
+    handleFocus,
+    handleBlur,
+    values,
+    errors,
+    activeFields,
+    touched,
+  );
+
+  const pickerProps = usePickerProps(
+    handleValueChange,
+    handleFocus,
+    handleBlur,
+    values,
+    errors,
+    activeFields,
+    touched,
+  );
+  const { onValueChange } = pickerProps(FIELDS.topic);
+
+  const isLoading = status === LOADING;
+
+  return (
+    <>
+      <Input
+        label={strings.CREATE_TARGET.areaLenght}
+        keyboardType="numeric"
+        autoCapitalize="none"
+        testID="area-lenght-input"
+        styleContainer={styles.inputContainer}
+        styleInput={styles.areaLenghtInput}
+        {...inputProps(FIELDS.areaLenght)}
```

Comment:
did you test if the user uses a decimal number? maybe you can parse `Math.Ceiling`

Reply:
Yes, it's working fine. The Target api allow decimal number and also in createTarget action the areaLenght is converted to number.